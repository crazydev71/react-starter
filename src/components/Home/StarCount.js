import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { testActions } from 'store/actions';

const StarCount = ({ testData, getTestData }) => {
  useEffect(() => {
    getTestData();
  }, [getTestData]);

  return (
    <div className="star-count">
      <h2>*** Example for using redux ***</h2>
      <h3>CRA repository start count: {testData.stargazers_count}</h3>
    </div>
  );
};

StarCount.propTypes = {
  testData: PropTypes.object,
  getTestData: PropTypes.func,
};

StarCount.defaultProps = {
  testData: {},
  getTestData: () => {},
};

const mapStateToProps = state => ({
  testData: state.test.testData,
});

const mapDispatchToProps = dispatch => ({
  getTestData: payload => dispatch(testActions.getTestDataRequest(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(StarCount);
