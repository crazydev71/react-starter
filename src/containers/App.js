import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { Switch, Route } from 'react-router-dom';

import configureStore, { history } from 'store/configure';
import Home from 'containers/Home';
import Page404 from 'containers/Page404';

import logo from 'static/img/logo.svg';

const store = configureStore();

function App() {
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <div className="App">
          <header className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
          </header>
          <main>
            <Switch>
              <Route path="/" exact component={Home} />
              <Route component={Page404} />
            </Switch>
          </main>
        </div>
      </ConnectedRouter>
    </Provider>
  );
}

export default App;
