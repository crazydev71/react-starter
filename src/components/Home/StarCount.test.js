import React from 'react';
import { shallow } from 'enzyme';
import App from 'App';
import StarCount from './StarCount';

describe('StarCount component', () => {
  it('Should render StarCount component.', () => {
    const wrapper = shallow(<App><StarCount /></App>);
    expect(wrapper).toMatchSnapshot();
  });
});

jest.setTimeout(20 * 1000);
