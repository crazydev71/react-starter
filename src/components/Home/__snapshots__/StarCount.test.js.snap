// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarCount component Should render StarCount component. 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "sagaTask": Object {
        "@@redux-saga/TASK": true,
        "cancel": [Function],
        "cont": [Function],
        "context": Object {},
        "end": [Function],
        "error": [Function],
        "id": 1,
        "isAborted": [Function],
        "isCancelled": [Function],
        "isRoot": true,
        "isRunning": [Function],
        "joiners": Array [],
        "meta": Object {
          "location": undefined,
          "name": "rootSaga",
        },
        "queue": Object {
          "abort": [Function],
          "addTask": [Function],
          "cancelAll": [Function],
          "getTasks": [Function],
        },
        "result": [Function],
        "setContext": [Function],
        "toPromise": [Function],
      },
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ConnectedRouterWithContext)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <div
      className="App"
    >
      <header
        className="App-header"
      >
        <img
          alt="logo"
          className="App-logo"
          src="logo.svg"
        />
      </header>
      <main>
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
          />
        </Switch>
      </main>
    </div>
  </Connect(ConnectedRouterWithContext)>
</Provider>
`;
